internal/core/app_deploy.go

```diff
@@ -339,15 +340,12 @@ func (op *deployOperation) Do(ctx context.Context, log hclog.Logger, app *App, m
 
 	// Convert from the plugin declaredResources to server declaredResources. Should be identical.
 	declaredResources := make([]*pb.DeclaredResource, len(declaredResourcesResp.DeclaredResources))
-	for i, resource := range declaredResourcesResp.DeclaredResources {
-		declaredResources[i] = &pb.DeclaredResource{
-			Id:                  resource.Id,
-			Name:                resource.Name,
-			Platform:            resource.Platform,
-			State:               resource.State,
-			StateJson:           resource.StateJson,
-			CategoryDisplayHint: pb.ResourceCategoryDisplayHint(resource.CategoryDisplayHint.Number()),
+	for i, pluginDeclaredResource := range declaredResourcesResp.DeclaredResources {
+		var serverDeclaredResource pb.DeclaredResource
+		if err := mapstructure.Decode(pluginDeclaredResource, &serverDeclaredResource); err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to decode plugin declared resource named %q: %s", pluginDeclaredResource.Name, err)
 		}
+		declaredResources[i] = &serverDeclaredResource
```

Comment:
Unrelated change - mapstructure is just a much nicer way to do this conversion.

Reply:
Nit: I think using mapstructure here makes the code _look_ nicer but I'm not convinced it _is_ nicer... the previous code is dead simple to understand, and wasn't doing any kind of conversions or manipulations of data types that would otherwise be handled by mapstructure for us (which is what mapstructure does really well). 

It also introduces an error scenario (however unlikely), which could/would be presented to the user and possibly be confusing as to what `mapstructure` is or doing. I admit that it's probably unlikely though.

Just my thoughts on the introduction/usage here